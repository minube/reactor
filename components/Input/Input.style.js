import { Platform, StyleSheet } from 'react-native';

import { LAYOUT, THEME } from '../../common';

const {
  COLOR, FONT, UNIT, SPACE,
} = THEME;

const ICON_SIZE = UNIT * 2;
const INPUT_HEIGHT = UNIT * 4.2;

export { INPUT_HEIGHT };

export default StyleSheet.create({
  container: {
    marginBottom: SPACE.REGULAR,
  },

  content: {
    ...LAYOUT.STYLE.ROW,
    paddingHorizontal: SPACE.XXS,
    borderColor: COLOR.BASE,
    borderWidth: 1,
    borderRadius: UNIT / 4,
  },

  disabled: {
    backgroundColor: COLOR.DISABLED,
  },

  error: {
    borderColor: COLOR.ERROR,
  },

  focus: {
    borderColor: COLOR.PRIMARY,
  },

  icon: {
    marginHorizontal: SPACE.XXS,
    height: ICON_SIZE,
    width: ICON_SIZE,
  },

  input: {
    backgroundColor: COLOR.TRANSPARENT,
    borderWidth: 0,
    color: COLOR.TEXT,
    flex: 1,
    fontSize: SPACE.MEDIUM,
    minHeight: INPUT_HEIGHT,
    paddingHorizontal: SPACE.XXS,
    paddingVertical: UNIT * 0.75,
    width: '100%',
    ...Platform.select({
      web: {
        outline: 'none',
      },
    }),
    ...FONT.INPUT,
  },

  inputDisabled: {
    color: COLOR.TEXT_LIGHTEN,
  },

  multiline: {
    paddingVertical: SPACE.XXS,
  },
});
