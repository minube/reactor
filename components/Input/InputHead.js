import React from 'react';

import Text from '../Text';
import styles from './InputHead.style';
import { string } from 'prop-types';

const InputHead = ({ title, ...inherit }) => (
    <Text {...inherit} numberOfLines={2} subtitle level={3} lighten style={styles.container}>{title}</Text>
);

InputHead.propTypes = {
    title: string,
};

export default InputHead;
