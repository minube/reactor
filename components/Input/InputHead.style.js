import { StyleSheet } from 'react-native';

import { THEME } from '../../common';

const { SPACE } = THEME;

export default StyleSheet.create({
  container: {
    marginTop: SPACE.XS,
    marginBottom: SPACE.L,
    paddingLeft: 0,
    maxWidth: '100%',
  },
});
