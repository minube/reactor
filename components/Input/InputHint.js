import React from 'react';

import Text from '../Text';
import styles from './InputHint.style';

const InputHint = inherit => (
  <Text {...inherit} caption level={2} lighten style={styles.container} />
);

export default InputHint;
