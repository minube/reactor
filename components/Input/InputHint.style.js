import { StyleSheet } from 'react-native';

import { THEME } from '../../common';

const { UNIT } = THEME;

export default StyleSheet.create({
  container: {
    marginTop: UNIT / 4,
    maxWidth: '100%',
  },
});
