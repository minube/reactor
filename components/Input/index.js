export { default, InputHint, InputLabel } from './Input';
