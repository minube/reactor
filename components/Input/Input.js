import {
  bool, func, oneOfType, number, string,
} from 'prop-types';
import React, { Component } from 'react';
import { TextInput, View } from 'react-native';

import { THEME } from '../../common';

import InputHint from './InputHint';
import InputLabel from './InputLabel';
import InputIcon from './InputIcon';
import Icon from '../Icon';
import Text from '../Text';
import styles from './Input.style';

const { COLOR } = THEME;

class Input extends Component {
  static propTypes = {
    color: string,
    currency: string,
    disabled: bool,
    error: oneOfType([bool, string]),
    hint: string,
    icon: oneOfType([number, string]),
    keyboard: string,
    label: string,
    lines: number,
    onBlur: func,
    onChange: func,
    onFocus: func,
    required: bool,
    requiredIcon: bool,
    valid: bool,
  };

  static defaultProps = {
    color: undefined,
    currency: undefined,
    disabled: false,
    error: undefined,
    hint: undefined,
    icon: undefined,
    keyboard: 'default',
    label: undefined,
    lines: undefined,
    onBlur: undefined,
    onChange: undefined,
    onFocus: undefined,
    required: false,
    requiredIcon: false,
    valid: false,
  };

  state = {
    focus: false,
  };

  render() {
    const {
      props: {
        color, currency, disabled, error, hint, icon, label, lines, required, requiredIcon, valid,
        onBlur, onChange, onFocus,
        ...inherit
      },
      state: { focus },
    } = this;
    let { props: { keyboard } } = this;
    if (currency) keyboard = 'numeric';

    return (
      <View style={[styles.container, inherit.style]}>
        { label && <InputLabel>{label}</InputLabel> }
        <View
          style={[
            styles.content,
            disabled && styles.disabled,
            !disabled && valid && styles.valid,
            !disabled && focus && (color ? { borderColor: color } : styles.focus),
            !disabled && error && styles.error,
          ]}
        >
          { (icon || currency) && (
            <View style={styles.inlineHint} pointerEvents="none">
              { icon && <Icon value={icon} style={styles.icon} />}
              { currency && <Text input lighten _style={styles.inlineHint}>{currency}</Text> }
            </View>
          )}


          <TextInput
            {...inherit}
            value={inherit.value || ''}
            autoCorrect={false}
            autoCapitalize="none"
            blurOnSubmit
            editable={!disabled}
            keyboardType={keyboard}
            numberOfLines={lines}
            multiline={lines > 1}
            onChangeText={onChange}
            onBlur={onBlur || (() => !disabled && this.setState({ focus: false }))}
            onFocus={onFocus || (() => !disabled && this.setState({ focus: true }))}
            placeholderTextColor={COLOR.TEXT_LIGHTEN}
            underlineColorAndroid="transparent"
            style={[
              styles.input,
              disabled && styles.inputDisabled,
              currency && styles.inputCurrency,
              inherit.fontSize && { fontSize: inherit.fontSize },
            ]}
          />
          { (error || (required && requiredIcon)) && (
            <Icon value={error ? 'error' : 'errorOutline'} style={[styles.icon, styles.iconRight]} />
          )}
          { valid && <InputIcon style={[styles.icon, styles.iconRight]} /> }
        </View>
        { hint && (
          <InputHint>
            {hint}
          </InputHint>
        )}
      </View>
    );
  }
}

export { InputHint, InputIcon, InputLabel };
export default Input;
