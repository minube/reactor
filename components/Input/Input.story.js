import React from 'react';
import { action } from '@storybook/addon-actions';
import { storiesOf } from '@storybook/react';
import { color, text, boolean, object, number, select } from '@storybook/addon-knobs/react';

import Input from './Input';

const STYLE = { backgroundColor: 'rgba(0,255,0,0.25)', padding: 10, width: 256 };

storiesOf('✅ Input', module)
  .add('default', () => (
    <Input />
  ))
  .add('color', () => (
    <Input color="green" />
  ))
  .add('disabled', () => (
    <Input disabled />
  ))
  .add('error', () => (
    <Input error="Required field" />
  ))
  .add('hint', () => (
    <Input hint="Optional fieldOptional fieldOptional fieldOptional fieldOptional fieldOptional field" />
  ))
  .add('keyboard (numeric)', () => (
    <Input keyboard="numeric" />
  ))
  .add('label', () => (
    <Input label="Username" />
  ))
  .add('lines', () => (
    <Input lines={5} />
  ))
  .add('value', () => (
    <Input value="soyjavi.com" />
  ))
  .add('⚡ onBlur', () => (
    <Input onBlur={action('Input.onBlur()')} title="Press me" />
  ))
  .add('⚡ onChange', () => (
    <Input onChange={action('Input.onChange()')} title="Press me" />
  ))
  .add('⚡ onFocus', () => (
    <Input onFocus={action('Input.onFocus()')} title="Press me" />
  ))
  .add('style', () => (
    <Input style={STYLE} />
  ))
  .add('...inherit', () => (
    <Input placeholder="placeholder..." />
  ))
  .add('🏀 Playground', () => (
    <Input
      color={color('color', undefined)}
      disabled={boolean('disabled', false)}
      error={text('error', null)}
      hint={text('hint', null)}
      keyboard={select('type', ['default', 'numeric', 'email-address', 'phone-pad'], 'default')}
      label={text('label', null)}
      lines={number('lines', 1)}
      value={text('value', null)}
      style={object('style', STYLE)}
    />
  ))

