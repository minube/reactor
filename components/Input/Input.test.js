import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import Input from './Input';

describe('<Input>', () => {
  it('renders', () => {
    const tree = renderer.create(<Input />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders in DOM environment', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Input />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
});

