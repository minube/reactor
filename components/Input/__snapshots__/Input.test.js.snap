// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input> renders 1`] = `
<View
  style={
    Array [
      Object {
        "marginBottom": 16,
        "minHeight": undefined,
      },
      undefined,
    ]
  }
>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    autoCorrect={false}
    blurOnSubmit={true}
    editable={true}
    keyboardType="default"
    multiline={false}
    numberOfLines={undefined}
    onBlur={[Function]}
    onChangeText={undefined}
    onFocus={[Function]}
    placeholderTextColor="#9b9b9b"
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#dbdbdb",
          "borderRadius": 2.5,
          "borderWidth": 1,
          "color": "#4a4a4a",
          "fontSize": 16,
          "fontWeight": "normal",
          "paddingHorizontal": 5,
          "paddingVertical": 7.5,
          "width": "100%",
        },
        false,
        false,
        undefined,
      ]
    }
    underlineColorAndroid="transparent"
    value=""
  />
</View>
`;

exports[`<Input> when {disabled} 1`] = `
<View
  style={
    Array [
      Object {
        "marginBottom": 16,
        "minHeight": undefined,
      },
      undefined,
    ]
  }
>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    autoCorrect={false}
    blurOnSubmit={true}
    editable={false}
    keyboardType="default"
    multiline={false}
    numberOfLines={undefined}
    onBlur={[Function]}
    onChangeText={undefined}
    onFocus={[Function]}
    placeholderTextColor="#9b9b9b"
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#dbdbdb",
          "borderRadius": 2.5,
          "borderWidth": 1,
          "color": "#4a4a4a",
          "fontSize": 16,
          "fontWeight": "normal",
          "paddingHorizontal": 5,
          "paddingVertical": 7.5,
          "width": "100%",
        },
        Object {
          "backgroundColor": "#f4f4f4",
          "color": "#cccccc",
        },
        false,
        false,
      ]
    }
    underlineColorAndroid="transparent"
    value=""
  />
</View>
`;

exports[`<Input> when {error} 1`] = `
<View
  style={
    Array [
      Object {
        "marginBottom": 16,
        "minHeight": undefined,
      },
      undefined,
    ]
  }
>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    autoCorrect={false}
    blurOnSubmit={true}
    editable={true}
    keyboardType="default"
    multiline={false}
    numberOfLines={undefined}
    onBlur={[Function]}
    onChangeText={undefined}
    onFocus={[Function]}
    placeholderTextColor="#9b9b9b"
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#dbdbdb",
          "borderRadius": 2.5,
          "borderWidth": 1,
          "color": "#4a4a4a",
          "fontSize": 16,
          "fontWeight": "normal",
          "paddingHorizontal": 5,
          "paddingVertical": 7.5,
          "width": "100%",
        },
        false,
        false,
        Object {
          "borderColor": "#ef5160",
        },
      ]
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    numberOfLines={undefined}
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
        },
        false,
        false,
        false,
        false,
        Object {
          "fontSize": 16,
          "lineHeight": 20.8,
        },
        Object {
          "fontSize": 12,
          "lineHeight": 14.399999999999999,
        },
        false,
        false,
        false,
        false,
        Object {
          "color": "#9b9b9b",
        },
        false,
        false,
        Object {
          "color": "#ef5160",
          "marginVertical": 2.5,
        },
      ]
    }
  >
    Hello World
  </Text>
</View>
`;

exports[`<Input> when {hint} 1`] = `
<View
  style={
    Array [
      Object {
        "marginBottom": 16,
        "minHeight": undefined,
      },
      undefined,
    ]
  }
>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    autoCorrect={false}
    blurOnSubmit={true}
    editable={true}
    keyboardType="default"
    multiline={false}
    numberOfLines={undefined}
    onBlur={[Function]}
    onChangeText={undefined}
    onFocus={[Function]}
    placeholderTextColor="#9b9b9b"
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#dbdbdb",
          "borderRadius": 2.5,
          "borderWidth": 1,
          "color": "#4a4a4a",
          "fontSize": 16,
          "fontWeight": "normal",
          "paddingHorizontal": 5,
          "paddingVertical": 7.5,
          "width": "100%",
        },
        false,
        false,
        undefined,
      ]
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    numberOfLines={undefined}
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
        },
        false,
        false,
        false,
        false,
        Object {
          "fontSize": 16,
          "lineHeight": 20.8,
        },
        Object {
          "fontSize": 12,
          "lineHeight": 14.399999999999999,
        },
        false,
        false,
        false,
        false,
        Object {
          "color": "#9b9b9b",
        },
        false,
        false,
        Object {
          "marginVertical": 2.5,
        },
      ]
    }
  >
    Hello World
  </Text>
</View>
`;

exports[`<Input> when {keyboard} 1`] = `
<View
  style={
    Array [
      Object {
        "marginBottom": 16,
        "minHeight": undefined,
      },
      undefined,
    ]
  }
>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    autoCorrect={false}
    blurOnSubmit={true}
    editable={true}
    keyboardType="numeric"
    multiline={false}
    numberOfLines={undefined}
    onBlur={[Function]}
    onChangeText={undefined}
    onFocus={[Function]}
    placeholderTextColor="#9b9b9b"
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#dbdbdb",
          "borderRadius": 2.5,
          "borderWidth": 1,
          "color": "#4a4a4a",
          "fontSize": 16,
          "fontWeight": "normal",
          "paddingHorizontal": 5,
          "paddingVertical": 7.5,
          "width": "100%",
        },
        false,
        false,
        undefined,
      ]
    }
    underlineColorAndroid="transparent"
    value=""
  />
</View>
`;

exports[`<Input> when {label} 1`] = `
<View
  style={
    Array [
      Object {
        "marginBottom": 16,
        "minHeight": undefined,
      },
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    numberOfLines={undefined}
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
        },
        false,
        false,
        false,
        false,
        Object {
          "fontSize": 16,
          "lineHeight": 20.8,
        },
        Object {
          "fontSize": 12,
          "lineHeight": 14.399999999999999,
        },
        false,
        false,
        false,
        false,
        Object {
          "color": "#9b9b9b",
        },
        false,
        false,
        Object {
          "marginVertical": 2.5,
        },
      ]
    }
  >
    Hello World
  </Text>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    autoCorrect={false}
    blurOnSubmit={true}
    editable={true}
    keyboardType="default"
    multiline={false}
    numberOfLines={undefined}
    onBlur={[Function]}
    onChangeText={undefined}
    onFocus={[Function]}
    placeholderTextColor="#9b9b9b"
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#dbdbdb",
          "borderRadius": 2.5,
          "borderWidth": 1,
          "color": "#4a4a4a",
          "fontSize": 16,
          "fontWeight": "normal",
          "paddingHorizontal": 5,
          "paddingVertical": 7.5,
          "width": "100%",
        },
        false,
        false,
        undefined,
      ]
    }
    underlineColorAndroid="transparent"
    value=""
  />
</View>
`;

exports[`<Input> when {lines} 1`] = `
<View
  style={
    Array [
      Object {
        "marginBottom": 16,
        "minHeight": undefined,
      },
      undefined,
    ]
  }
>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    autoCorrect={false}
    blurOnSubmit={true}
    editable={true}
    keyboardType="default"
    multiline={true}
    numberOfLines={5}
    onBlur={[Function]}
    onChangeText={undefined}
    onFocus={[Function]}
    placeholderTextColor="#9b9b9b"
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#dbdbdb",
          "borderRadius": 2.5,
          "borderWidth": 1,
          "color": "#4a4a4a",
          "fontSize": 16,
          "fontWeight": "normal",
          "paddingHorizontal": 5,
          "paddingVertical": 7.5,
          "width": "100%",
        },
        false,
        false,
        undefined,
      ]
    }
    underlineColorAndroid="transparent"
    value=""
  />
</View>
`;

exports[`<Input> when {style} 1`] = `
<View
  style={
    Array [
      Object {
        "marginBottom": 16,
        "minHeight": undefined,
      },
      Object {
        "backgroundColor": "orange",
        "padding": 10,
        "width": 256,
      },
    ]
  }
>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    autoCorrect={false}
    blurOnSubmit={true}
    editable={true}
    keyboardType="default"
    multiline={false}
    numberOfLines={undefined}
    onBlur={[Function]}
    onChangeText={undefined}
    onFocus={[Function]}
    placeholderTextColor="#9b9b9b"
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#dbdbdb",
          "borderRadius": 2.5,
          "borderWidth": 1,
          "color": "#4a4a4a",
          "fontSize": 16,
          "fontWeight": "normal",
          "paddingHorizontal": 5,
          "paddingVertical": 7.5,
          "width": "100%",
        },
        false,
        false,
        undefined,
      ]
    }
    underlineColorAndroid="transparent"
    value=""
  />
</View>
`;

exports[`<Input> when {value} 1`] = `
<View
  style={
    Array [
      Object {
        "marginBottom": 16,
        "minHeight": undefined,
      },
      undefined,
    ]
  }
>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    autoCorrect={false}
    blurOnSubmit={true}
    editable={true}
    keyboardType="default"
    multiline={false}
    numberOfLines={undefined}
    onBlur={[Function]}
    onChangeText={undefined}
    onFocus={[Function]}
    placeholderTextColor="#9b9b9b"
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#dbdbdb",
          "borderRadius": 2.5,
          "borderWidth": 1,
          "color": "#4a4a4a",
          "fontSize": 16,
          "fontWeight": "normal",
          "paddingHorizontal": 5,
          "paddingVertical": 7.5,
          "width": "100%",
        },
        false,
        false,
        undefined,
      ]
    }
    underlineColorAndroid="transparent"
    value="soyjavi"
  />
</View>
`;
